packages/grid/x-data-grid/src/components/base/GridErrorHandler.tsx

```diff
@@ -1,35 +1,27 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import { useGridPrivateApiContext } from '../../hooks/utils/useGridPrivateApiContext';
-import { useGridLogger } from '../../hooks/utils/useGridLogger';
 import { GridMainContainer } from '../containers/GridMainContainer';
-import { ErrorBoundary } from '../ErrorBoundary';
 import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
 
 function GridErrorHandler(props: { children: React.ReactNode }) {
   const { children } = props;
-  const apiRef = useGridPrivateApiContext();
-  const logger = useGridLogger(apiRef, 'GridErrorHandler');
   const rootProps = useGridRootProps();
-  const errorState = apiRef.current.state.error;
+  const error = rootProps.error;
```

Comment:
Do you mean renaming the prop to something more generic?

Reply:
> For instance, developers may want to still show the rows but block any interaction until some other task is finished

I think it's way easier to do this outside of the grid:
```tsx
<div style={{ width: '100%', position: 'relative' }}>
  <DataGrid autoHeight {...data} />
  <div style={{ position: 'absolute', top: 0, width: '100%', height: '100%' }}>
    overlay
  </div>
</div>
```
https://codesandbox.io/s/hardcore-leavitt-yzd4ij?file=/demo.tsx

I wonder if the `error` prop is useful at all at this point.
What do you think?

Reply:
> the same ErrorOverlay component will be used for internal and user errors

After removing the internal error boundary, there will be no errors caught by the grid itself.
The `ErrorOverlay` will only be used when the user sets the `error` prop.

> If I had to show a error message, not originated by the grid, I would prefer an overlay taking the entire screen or a toast, instead of hidding the rows.

Agree. I really don't see much of a value provided by the ErrorOverlay at this point.